lib/yocaml/key_value.mli

```diff
@@ -0,0 +1,334 @@
+(** To attach additional content to documents, [YOCaml] uses a Metadata
+    mechanism. Generally, the formats used for this kind of metadata ([Yaml],
+    [JSON] or even [TOML]) can be abstractly represented as abstract objects,
+    like a key-value table.
+
+    This module abstracts the validation logic for key-value structured data. *)
+
+(** {1 Description of a format}
+
+    To be able to validate structured data, one must first provide visitors
+    for each data type supported by the metadata description logic. *)
+
+(** A visitor takes a structured data structure and if it respects the
+    expected schema, applies a function with the extracted data, otherwise it
+    applies another function. In a slightly less generic way, the function
+    that acts on the extracted data can act as a validation and the one that
+    is applied in case of non respect of the structure is the failure
+    function.*)
+type ('a, 'b, 'c) visitor = ('b -> 'c) -> (unit -> 'c) -> 'a -> 'c
+
+(** {2 The set of visitors for a key-value structurable object} *)
+
+(** Creating a set of validators for a key-value structure only involves
+    implementing the {!module-type:KEY_VALUE_OBJECT} module. *)
+module type KEY_VALUE_OBJECT = sig
```

Comment:
Since we are already in the module `Key_value`, do you think it is necessary to prefix signatures with `KEY_VALUE`?